keyvault/client/src/main/java/com/azure/keyvault/SecretClient.java

```diff
@@ -0,0 +1,540 @@
+    }
+
+    /**
+     * Creates a builder that can configure options for the SecretClient before creating an instance of it.
+     * @return A new Builder to create a SecretClient from.
+     */
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    /**
+     * Provides configuration options for instances of {@link SecretClient}.
+     */
+    public static final class Builder {
+        private final List<HttpPipelinePolicy> policies;
+        private ServiceClientCredentials credentials;
+        private String vaultEndPoint;
+        private HttpClient httpClient;
+        private HttpLogDetailLevel httpLogDetailLevel;
+        private RetryPolicy retryPolicy;
+        private String userAgent;
+
+        private Builder() {
+            userAgent = String.format("Azure-SDK-For-Java/%s (%s)", SDK_NAME, SDK_VERSION);
+            retryPolicy = new RetryPolicy();
+            httpLogDetailLevel = HttpLogDetailLevel.NONE;
+            policies = new ArrayList<>();
+        }
+
+        /**
+         * Creates a {@link SecretClient} based on options set in the Builder.
+         *
+         * Every time {@code build()} is called, a new instance of {@link SecretClient} is created.
+         *
+         * @return A SecretClient with the options set from the builder.
+         * @throws IllegalStateException If {@link Builder#credentials(ServiceClientCredentials)}
+         * @throws IllegalStateException If {@link Builder#credentials(ServiceClientCredentials)}
+         * has not been set.
+         */
+        public SecretClient build() {
+            if (credentials == null) {
+                throw new IllegalStateException("'credentials' is required.");
+            }
+
+            if (vaultEndPoint == null){
+                throw new IllegalStateException("'Vault's Endpoint Url' is required.");
+            }
+
+            // Closest to API goes first, closest to wire goes last.
+            final List<HttpPipelinePolicy> policies = new ArrayList<>();
+
+            policies.add(new UserAgentPolicy(userAgent));
+            policies.add(new RequestIdPolicy());
+            policies.add(retryPolicy);
+            policies.add(new KeyvaultCredentialsPolicy(credentials, vaultEndPoint));
+
+            policies.addAll(this.policies);
+
+            policies.add(new HttpLoggingPolicy(httpLogDetailLevel));
+
+            HttpPipeline pipeline = httpClient == null
+                    ? new HttpPipeline(policies)
+                    : new HttpPipeline(httpClient, policies);
+
+            return new SecretClient(vaultEndPoint, pipeline);
+        }
+
+        /**
+         * Sets the credentials to use when authenticating HTTP requests.
+         *
+         * @param vaultEndPoint The vault endpoint url is used as destination on Azure to send requests to.
+         * @return The updated Builder object.
+         * @throws NullPointerException if {@code credentials} is {@code null}.
```

Comment:
copy paste fail....a few mentions of 'credentials'

Reply:
fixed.